From defee6759e59d79f401740eac8bd687e9ae6871b Mon Sep 17 00:00:00 2001
From: Phil Steitz <psteitz@apache.org>
Date: Sat, 23 Apr 2011 23:54:44 +0000
Subject: [PATCH] Changed DelegatingDatabaseMetaData to no longer add itself to
 the AbandonedTrace of its parent connection.  This was causing excessive
 memory consumption and was not necessary, as resultsets created by
 DelegatingDatabaseMetaData instances are attached to the parent connection's
 trace on creation. JIRA: DBCP-330 JIRA: DBCP-352

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/dbcp/branches/DBCP_1_4_x_BRANCH@1096259 13f79535-47bb-0310-9956-ffa450edef68
---
 src/changes/changes.xml                         |  6 ++++++
 .../dbcp/DelegatingDatabaseMetaData.java        | 17 +++++++----------
 2 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index dace780c94..52d6f61912 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -39,6 +39,12 @@ The <action> type attribute can be add,update,fix,remove.
   </properties>
   <body>
     <release version="1.4.1" date="TBD" description="TBD">
+      <action dev="psteitz" issue="DBCP-330" type="fix">
+        Changed DelegatingDatabaseMetaData to no longer add itself to the AbandonedTrace
+        of its parent connection.  This was causing excessive memory consumption and was
+        not necessary, as resultsets created by DelegatingDatabaseMetaData instances are
+        attached to the parent connection's trace on creation.  Also fixes DBCP-352.
+      </action>
       <action dev="psteitz" issue="DBCP-343" type="fix">
         Modified execute methods of Statement objects to ensure that whenever
         a statement is used, the lastUsed property of its parent connection is
diff --git a/src/java/org/apache/commons/dbcp/DelegatingDatabaseMetaData.java b/src/java/org/apache/commons/dbcp/DelegatingDatabaseMetaData.java
index df880469c9..cc3ccd4ff6 100644
--- a/src/java/org/apache/commons/dbcp/DelegatingDatabaseMetaData.java
+++ b/src/java/org/apache/commons/dbcp/DelegatingDatabaseMetaData.java
@@ -26,17 +26,14 @@
 import java.sql.SQLException;
 
 /**
- * A base delegating implementation of {@link DatabaseMetaData}.
- * <p>
- * Those methods that create {@link ResultSet} objects, are wrapped to
+ * <p>A base delegating implementation of {@link DatabaseMetaData}.</p>
+ * 
+ * <p>Methods that create {@link ResultSet} objects are wrapped to
  * create {@link DelegatingResultSet} objects and the remaining methods
  * simply call the corresponding method on the "delegate"
- * provided in my constructor.
- * <p>
- * Extends AbandonedTrace to implement DatabaseMetaData tracking and
- * logging of code which created the DatabaseMetaData. Tracking
- * the DatabaseMetaData ensures that the Connection which created it can
- * close any associated ResultSets on Connection close.
+ * provided in the constructor.</p>
+ * 
+ * <p>NOTE: as of version 2.0, this class will no longer extend AbandonedTrace.</p>
  */
 public class DelegatingDatabaseMetaData extends AbandonedTrace
         implements DatabaseMetaData {
@@ -49,7 +46,7 @@ public class DelegatingDatabaseMetaData extends AbandonedTrace
 
     public DelegatingDatabaseMetaData(DelegatingConnection c,
             DatabaseMetaData m) {
-        super(c);
+        super();
         _conn = c;
         _meta = m;
     }
